src/pages/HousePage/HousePage.tsx

```diff
@@ -1,14 +1,26 @@
 import { useParams } from "react-router-dom";
 import { useGetOneHouseQuery } from "../../redux/api/housesApi";
+import { useAppDispatch } from "../../hooks/hooks";
+import { toggleFavorite } from "../../redux/slices/userSlice";
+import { useCurrentUser } from "../../hooks/hooks";
+import FavoriteBorderIcon from '@mui/icons-material/FavoriteBorder';
+import FavoriteIcon from '@mui/icons-material/Favorite';
 import './HousePage.css';
 
 export const HousePage = () => {
-  const params = useParams<{ name?: string }>();
-  const { data } = useGetOneHouseQuery(params.name as string);
+  const dispatch = useAppDispatch();
+  const paramsName: string = useParams().name || '';
```

Comment:
```suggestion
  const { name: userParamValue = '' } = useParams();
```
Можно тип для useParams задать, что name там ожидаем, если хочется строгости ещё.